mlflow/tracking/client.py

```diff
@@ -638,16 +639,17 @@ def end_trace(
         root_span_id = trace_manager.get_root_span_id(request_id)
 
         if root_span_id is None:
-            if self.get_trace(request_id=request_id):
-                raise MlflowException(
-                    f"Trace with ID {request_id} already finished.",
-                    error_code=INVALID_PARAMETER_VALUE,
-                )
-            else:
+            trace = self.get_trace(request_id=request_id)
+            if trace is None:
                 raise MlflowException(
                     f"Trace with ID {request_id} not found.",
                     error_code=RESOURCE_DOES_NOT_EXIST,
                 )
+            elif trace.info.status == TraceStatus.OK or trace.info.status == TraceStatus.ERROR:
```

Comment:
super-nit: Perhaps `!= TraceStatus.IN_PROGRESS` is better because we may add more end state for trace (slightly more likely than adding another pending status)